### ISSUE TITLE
linting: Check format_source argument is a defined input file

### ISSUE THREAD

**Issue by peterjc** [galaxyproject/planemo#144]
e.g. This would have caught this error of mine:

https://github.com/peterjc/pico_galaxy/commit/734d298ac56af21380e92b3985ca971fc73e577b

I was replacing `format="input"` given the planemo warning:

```
.. WARNING: Using format='input' on output data, format_source attribute is less ambiguous and should be used instead.
```

**Comment by jmchilton** [comment]
Thanks - sure this makes sense - it should validate ``metadata_source`` as well.

**Comment by peterjc** [comment]
Yes, applying the same validation on ``metadata_source`` makes sense.

**Comment by jmchilton** [comment]
From my agent:

> This is implemented upstream in galaxy-tool-util, which is where planemo's tool linting now
> lives: the `OutputsFormatSourceReference` linter in `lib/galaxy/tool_util/linters/output.py`
> reports
> `Output '<name>' references format_source='<value>' which does not match any input parameter.`
> as an ERROR, and stays quiet for a valid reference. Verified by running
> `planemo lint` (planemo 0.75.45.dev0, galaxy-tool-util 26.0.1) against a tool with
> `format_source="nosuchinput"`, including with a legacy tool profile. Two narrower gaps remain:
> `metadata_source` is not validated at all, and the `format_source` check confirms only that a
> param of that name exists, not that it is a data input. Those would be better tracked in a
> separate issue than kept on this one.

-John
